network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/NetPVpnSettingsViewModel.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.impl.settings
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.duckduckgo.anvil.annotations.ContributesViewModel
+import com.duckduckgo.app.global.DispatcherProvider
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.networkprotection.impl.settings.geoswitching.DisplayablePreferredLocationProvider
+import javax.inject.Inject
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+@ContributesViewModel(ActivityScope::class)
+class NetPVpnSettingsViewModel @Inject constructor(
+    private val dispatcherProvider: DispatcherProvider,
+    private val displayablePreferredLocationProvider: DisplayablePreferredLocationProvider,
+) : ViewModel() {
+    private val _viewState = MutableStateFlow(ViewState())
+    internal fun viewState(): Flow<ViewState> = _viewState.asStateFlow()
+
+    internal data class ViewState(
+        val preferredLocation: String? = null,
+    )
+
+    internal fun initialize() {
+        viewModelScope.launch(dispatcherProvider.io()) {
+            displayablePreferredLocationProvider.getDisplayablePreferredLocation().also {
+                _viewState.emit(_viewState.value.copy(preferredLocation = it))
+            }
+        }
+    }
```

Comment:
What about using the lifecycle of the activity instead of this `initialize` method the activity needs to proactively call and more importantly, know when to call?

Reply:
I’ve fixed this in https://github.com/duckduckgo/Android/pull/3673 which is where this PR is stacked from.